src/redux/contact/tests/contact.sagas.test.js

```diff
@@ -30,45 +36,158 @@ import {
+        const analysisCall = analysis.filter((e) => e.type === 'CALL');
+        expect(analysisPut).toHaveLength(4);
+        expect(analysisCall).toHaveLength(2);
+      }));
 
-  it('Should delete contact and remove it from store', () => {
-    expectSaga(handleContactDelete, contactId)
-      .provide([[matchers.call.fn(deleteContact), contactDeleteIdMock]])
+  it('Should delete contact and remove it from store', () =>
+    expectSaga(handleContactDelete, { payload: contact._id })
+      .withReducer(contactReducer)
+      .provide([
+        [call(deleteContact, contact._id)],
+        [call(getContacts, payload.skip, payload.limit), contactRes],
+        [call(handleSuccessSnackbar, SUCCESS_DELETE_STATUS)]
+      ])
       .put(setContactsLoading(true))
-      .put(setContact(contactDeleteIdMock))
-      .put(deleteContactInStore(contactId))
-      .put(setContactsPagesCount(contactsPagesCount - 1))
+      .put(deleteContactInStore(contact._id))
+      .put(
+        setContactsPagesCount(
+          Math.ceil(contactRes.count / payload.contactsPerPage)
+        )
+      )
+      .put(setContacts(contactRes.items))
+      .put(setContactsLoading(false))
+      .hasFinalState({
+        ...initialState,
+        contacts: contactRes.items
+      })
+      .run()
+      .then((result) => {
+        const { allEffects: analysis } = result;
+        const analysisPut = analysis.filter((e) => e.type === 'PUT');
+        const analysisCall = analysis.filter((e) => e.type === 'CALL');
+        expect(analysisPut).toHaveLength(6);
+        expect(analysisCall).toHaveLength(3);
+      }));
+
+  it('Should update contact', () =>
+    expectSaga(handleContactUpdate, {
+      payload: { id: contact._id, updatedContact, mapImages: [] }
+    })
+      .withReducer(contactReducer)
+      .provide([
+        [call(updateContact, contact._id, updatedContact, [])],
+        [call(handleSuccessSnackbar, SUCCESS_UPDATE_STATUS)]
+      ])
+      .put(setContactsLoading(true))
+      .put(updateContactInStore(contact._id, updatedContact))
+      .put(push('/contacts'))
+      .hasFinalState({
+        ...initialState,
+        contactsLoading: true
+      })
+      .run()
+      .then((result) => {
+        const { allEffects: analysis } = result;
+        const analysisPut = analysis.filter((e) => e.type === 'PUT');
+        const analysisCall = analysis.filter((e) => e.type === 'CALL');
+        expect(analysisPut).toHaveLength(3);
+        expect(analysisCall).toHaveLength(2);
+      }));
+
+  it('Should handle error', () =>
+    expectSaga(handleContactsError, error)
+      .withReducer(contactReducer)
+      .provide([[call(handleSuccessSnackbar, error.message)]])
       .put(setContactsLoading(false))
-      .run();
-  });
+      .put(setContactsError({ e: error }))
+      .hasFinalState({
+        ...initialState,
+        contactsLoading: false,
+        contactsError: { e: { message: error.message } }
```

Comment:
You will have conflict with fix-comment-tests branch

Reply:
solved